### DISCUSSION TITLE
install from ruby checkout on local machine

### DISCUSSION THREAD

**Discussion by fables-tales** [postmodern/ruby-install#493] [Q&A]
as far as I can tell, there isn't a way to provide ruby-install with a directory path to a ruby checkout on my machine. It'd be nice to have it be able to install a ruby from a working directory so that I can test out patches in a more long term. Is this a use case that could be supported?

**Comment by havenwood** [comment]
It's not directly supported, but you can pair `--src-dir` with `--no-extract` to effectively install a Ruby from a working directory if the directory is named as expected. For example, you can install current trunk with:

```sh
git clone git@github.com:ruby/ruby.git ruby-2.7.0dev --depth 1
ruby-install --no-extract --src-dir . ruby-2.7.0dev
```

There currently isn't a flag that sets the `$src_dir/$ruby_dir_name` directly for an arbitrarily-named `$ruby_dir_name`.

**Comment by havenwood** [comment]
A possibility would be to add a `--ruby-src-dir` option. Maybe something like:

```
-S, --ruby-src-dir DIR         Source code directory of a particular Ruby
```

**Comment by postmodern** [comment]
Part of `ruby-install`'s philosophy is that it explicitly does not support installing from trunk/master. Although, I am considering making a separate utility called `ruby-dev` for cloning and building development versions of rubies. If you only want to apply patches to the current version, than you can specify the `-p,--patch FILE` option to apply additional patches to the extracted Ruby's source code before compilation.
